### PR TITLE
POC - use select box table in hardware modals

### PR DESCRIPTION
**Pull request by MariaAga**
Shows a use in hardware modals for the select boxes from: https://github.com/theforeman/foreman/pull/7679
Only shows selection, doesn't do anything with the selected modals

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/common/table/TableSelection.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+**selectedRows**: an array of the selected ids  
+**showSelectAll**: a boolean that indicates should the select all option be available?  
+This is true after a user is selection the whole page. Use this to show a select all button or message box.
+
+## Connect The Actions
+
+Create a `selectionController` using the `getSelectionController` helper
+
+```js
+getSelectionController({
```
Comment (xprazak2) [inline webpack/assets/javascripts/react_app/components/common/table/TableSelection.stories.mdx:29]: When hooking up multiple tables, the dispatched actions will have same types for different tables if I am reading this correctly. I am not sure if redux will be able to update only the desired part of the store that way...

**Comment by xprazak2** [comment]
Nice, the selection seems to work as expected:

selection

File: webpack/assets/javascripts/react_app/components/common/table/TableSelection.stories.mdx (hunk 1 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/common/table/TableSelection.stories.mdx:29]: Thanks for the feedback, maybe there can be general reducers and selectors like in the API

Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/common/table/TableSelection.stories.mdx:29]: Each selection action should have a table id to work now

**Comment by ezr-ondrej** [comment]
@MariaAga what is the status? Will you need some help to implement the background action for this? bulk delete would be a good start I guess. Or do we want to do it on different resource?

**Comment by MariaAga** [comment]
I don't have any plans for this as this was a POC to see how to use the select helper for the table.

**Comment by ezr-ondrej** [comment]
oki :+1: I'll poke it in the PR we wanted the helper for :)

**Pull request closed without merging**